### PR TITLE
python3Packages.coremltools: init at 4.1

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

New package for working with Machine Learning Models

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/coremltools/default.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    pillow
+    pytest
+    pytorch
+    scikitlearn
+    torchvision
+  ];
+
+  # skip tests which require downloaded models or macOS
+  checkPhase = ''
+    pytest -k "not (
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/coremltools/default.nix:60]: Please use pytestCheckHook together with disabledTests.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105834` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python37Packages.coremltools</li>
    <li>python39Packages.coremltools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python38Packages.coremltools</li>
  </ul>
</details>

File: pkgs/development/python-modules/coremltools/default.nix (hunk 1 above)
Comment (prusnak) [reply pkgs/development/python-modules/coremltools/default.nix]: Addressed in f7777b8c133b821e5098dae6a317be7ca2ebf724

**Comment by prusnak** [comment]
In f7777b8c133b821e5098dae6a317be7ca2ebf724 I disabled test for everything else except Python 3.8

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105834` [1](https://github.com/Mic92/nixpkgs-review) on darwin
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python37Packages.coremltools</li>
    <li>python38Packages.coremltools</li>
    <li>python39Packages.coremltools</li>
  </ul>
</details>

Basically all tests fail.

**Comment by SuperSandro2000** [comment]
> In [f7777b8](https://github.com/NixOS/nixpkgs/commit/f7777b8c133b821e5098dae6a317be7ca2ebf724) I disabled test for everything else except Python 3.8

Please fix the eval error. ``isPy38`` is undefined.

**Comment by prusnak** [comment]
> Please fix the eval error. `isPy38` is undefined.

Fixed in 54bc78099b4f7e8c89807a2d6812985e8678deb0.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105834` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.coremltools</li>
    <li>python38Packages.coremltools</li>
    <li>python39Packages.coremltools</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105834` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python38Packages.coremltools</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.coremltools</li>
    <li>python39Packages.coremltools</li>
  </ul>
</details>

**Pull request closed without merging**